src/Tools/ConfigurationSchemaGenerator/ConfigSchemaEmitter.cs

```diff
@@ -401,12 +401,21 @@ private void AppendTypeNodes(JsonObject propertyNode, TypeSpec propertyTypeSpec)
         }
     }
 
+    const string NegativeOption = @"-?";
+    const string DaysAlone = @"\d{1,7}";
+    const string DaysPrefixOption = @$"({DaysAlone}[\.:])?";
+    const string MinutesOrSeconds = @"[0-5]?\d";
+    const string HourMinute = @$"([01]?\d|2[0-3]):{MinutesOrSeconds}";
+    const string HourMinuteSecond = HourMinute + $":{MinutesOrSeconds}";
+    const string SecondsFractionOption = @"(\.\d{1,7})?";
+    internal const string TimeSpanRegex = $"^{NegativeOption}({DaysAlone}|({DaysPrefixOption}({HourMinute}|{HourMinuteSecond}){SecondsFractionOption}))$";
```

Comment:
FWIW [STJ is less permissive](https://github.com/eiriktsarpalis/stj-schema-mapper/commit/c8b323635b83f24f43ada4ccc6efc883c8c01a01#r140502915) compared to what `TimeSpan.Parse` permits. I wonder if it makes sense to make the two components consistent since they are somewhat related.

Reply:
I opened https://github.com/dotnet/runtime/issues/100538 for making STJ be consistent with other APIs, like Configuration Binder, TimeSpan.Parse, and `Utf8Parser.TryParse(..., out TimeSpan)`